indexer/rank_table.cpp

```diff
@@ -214,10 +220,29 @@ unique_ptr<RankTable> LoadRankTable(unique_ptr<TRegion> && region)
   return unique_ptr<RankTable>();
 }
 
+uint8_t CalcEventRank(FeatureType const & ft)
+{
+  uint8_t const fc2018Rank = 16;  // todo(@t.yan): test and adjust.
+  Classificator const & c = classif();
+  auto const types = feature::TypesHolder(ft);
+  auto const fcType =
+      ftypes::BaseChecker::PrepareToMatch(c.GetTypeByPath({"event", "fc2018"}), 2 /* level */);
+  auto const fcCityType =
+      ftypes::BaseChecker::PrepareToMatch(c.GetTypeByPath({"event", "fc2018_city"}), 2 /* level */);
+  if (find(types.begin(), types.end(), fcType) != types.end() ||
+      find(types.begin(), types.end(), fcCityType) != types.end())
+  {
+    return fc2018Rank;
+  }
```

Comment:
Not all paths from this function return.

Reply:
fixed